### ISSUE TITLE
Avalanche Warp Signing Block Hashes

### ISSUE THREAD

**Issue by aaronbuchwald** [ava-labs/subnet-evm#734]
**Context and scope**
Avalanche Warp Messaging currently provides the functionality to send a point to point message from chain A address A to chain B address B with an arbitrary byte array payload as specified here: https://github.com/ava-labs/subnet-evm/blob/master/contracts/contracts/interfaces/IWarpMessenger.sol#L8.

This means that Avalanche Warp Messages will only be signed by subnet validators after an explicit request is performed on-chain.

In some cases, Subnets may want to be configured to support signing block hashes either every block, after an explicit request, or every N blocks.

Why should Subnet-EVM sign block hashes?

The block hash attests to the contents of an EVM block including a state root, receipt root, and transaction root. Combined with a signature from an Avalanche validator set, the signed block hash can be verified via Warp in order to verify the state or execution result on the chain that signed the warp message.

Many existing cross-chain communication protocols leverage a signed block hash in order to prove everything that they need between two different blockchains (ex. https://layerzero.gitbook.io/docs/ecosystem/oracle/develop-an-oracle).

**Discussion and alternatives**
One alternative is to use the currently implemented point to point Avalanche Warp Messaging system to deliver a block hash where it's needed.

To implement this, you would currently need to deploy a contract in order to use the `BLOCKHASH` opcode and request a signature from the validator set via the Warp precompile.

However, the `BLOCKHASH` opcode only supports retrieving block hashes between 1 and 256 blocks old, but does not support getting the current block hash. This is because when building a block, it is not known what the resulting block hash will be.

Therefore, a system built relying on the `BLOCKHASH` opcode would increase the latency of sending a block hash from chain A to chain B via Warp by one additional block confirmation time. A simple solution to avoid adding this additional latency is to provide first class support for signing block hashes directly as opposed to only signing point to point messages.

**Initial Design Idea**

1. Add a new message type to the `Warp` message codec: https://github.com/ava-labs/subnet-evm/blob/master/warp/payload/codec.go#L25 to include only a block hash:

```
type BlockHashPayload struct {
	BlockHash      ids.ID `serialize:"true"`

	bytes []byte
}
```

2. Add support for the new message type to `IWarpMessenger.sol` https://github.com/ava-labs/subnet-evm/blob/master/contracts/contracts/interfaces/IWarpMessenger.sol#L8 and add a new function `getVerifiedWarpBlockHash` to retrieve and return the new message type

3. If warp is enabled, add the block hash to the Warp Backend during every block accept

**Comment by aaronbuchwald** [comment]
I think it would be better to add a separate message type for block hashes instead of trying to fit it into the `AddressedPayload`. Ideally, this can be used across different VMs, and I don't think it will make sense for every VM to force the format of signing a block hash into the address-to-address message format.

On the second point, that's a good idea, it should be fine to just add a block lookup instead of adding the same data into the warp backend as an unsigned message.

**Comment by aaronbuchwald** [comment]
Addressed by #823
